internal/browser/browser.go

```diff
@@ -26,11 +28,36 @@ type Opener interface {
 
 type StderrPrint struct{}
 
+func openURL(url string) error {
+	var err error
+	switch runtime.GOOS {
+	case "darwin": // macOS
+		err = exec.Command("open", url).Start()
+	case "linux":
+		providers := []string{"xdg-open", "x-www-browser", "www-browser"}
+
+		// There are multiple possible providers to open a browser on linux
+		// One of them is xdg-open, another is x-www-browser, then there's www-browser, etc.
+		// Look for one that exists and run it
+		for _, provider := range providers {
+			if _, err := exec.LookPath(provider); err == nil {
+				err = exec.Command(provider, url).Start()
+			}
+		}
+	case "windows":
+		err = exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start()
+	default:
+		err = fmt.Errorf("unsupported platform")
+	}
+	return err
+}
+
 func (p *StderrPrint) Open(u *url.URL) error {
 	fmt.Fprintf(
-		os.Stderr,
-		"Please open the following URL in your web browser to authenticate:\n\n\t%s\n\n",
+		os.Stdout,
+		"Opening the following URL in your web browser to authenticate:\n\n\t%s\n\n",
```

Comment:
The user might be running this command on a remote machine whether there is no browser or where they can't see the browser. I think it still makes sense to tell them to open the link manually in the case.

For reference, this is what `aws sso login` prints:

```
Attempting to automatically open the SSO authorization page in your default browser.
If the browser does not open or you wish to use a different device to authorize this request, open the following URL:

https://device.sso.us-east-2.amazonaws.com/

Then enter the code:
```